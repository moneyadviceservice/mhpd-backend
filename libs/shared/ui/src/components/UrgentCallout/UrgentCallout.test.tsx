import React from 'react';
import { render, screen } from '@testing-library/react';
import { UrgentCallout } from '.';

describe('UrgentCallout component', () => {
  it('renders with an arrow', () => {
    render(
      <UrgentCallout className="testClassName" variant="arrow">
        children
      </UrgentCallout>,
    );
    const callout = screen.getByTestId('urgent-callout');
    expect(callout).toMatchSnapshot();
  });

  it('renders with a calculator', () => {
    render(
      <UrgentCallout className="testClassName" variant="calculator">
        children
      </UrgentCallout>,
    );

    const callout = screen.getByTestId('urgent-callout');
    expect(callout).toMatchSnapshot();
  });

  it('renders with a warning', () => {
    render(
      <UrgentCallout className="testClassName" variant="warning">
        children
      </UrgentCallout>,
    );

    const callout = screen.getByTestId('urgent-callout');
    expect(callout).toMatchSnapshot();
  });

  it('renders with an arrow and teal border', () => {
    render(
      <UrgentCallout className="testClassName" variant="arrow" border="teal">
        children
      </UrgentCallout>,
    );
    const callout = screen.getByTestId('urgent-callout');
    expect(callout).toMatchSnapshot();
  });

  it('renders with a calculator and teal border', () => {
    render(
      <UrgentCallout
        className="testClassName"
        variant="calculator"
        border="teal"
      >
        children
      </UrgentCallout>,
    );

    const callout = screen.getByTestId('urgent-callout');
    expect(callout).toMatchSnapshot();
  });

  it('renders with a warning and teal border', () => {
    render(
      <UrgentCallout className="testClassName" variant="warning" border="teal">
        children
      </UrgentCallout>,
    );

    const callout = screen.getByTestId('urgent-callout');
    expect(callout).toMatchSnapshot();
  });

  it('adds className and style correctly', () => {
    render(
      <UrgentCallout
        variant="warning"
        style={{ color: 'blue' }}
        className="testClassName"
      >
        children
      </UrgentCallout>,
    );

    const callout = screen.getByTestId('urgent-callout');
    expect(callout).toMatchSnapshot();
  });
});
