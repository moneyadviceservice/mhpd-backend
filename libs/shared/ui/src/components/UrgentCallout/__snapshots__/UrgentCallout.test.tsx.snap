// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrgentCallout component adds className and style correctly 1`] = `
<div
  class="t-urgent-callout border-8 rounded-bl-3xl py-10 pl-6 pr-8 sm:p-10 flex border-yellow-200 testClassName"
  data-testid="urgent-callout"
  style="color: blue;"
>
  <div
    class="flex gap-4 sm:gap-6"
  >
    <div
      class="text-pink-800"
    >
      <svgr-mock
        classname=""
      />
    </div>
    <div>
      children
    </div>
  </div>
</div>
`;

exports[`UrgentCallout component renders with a calculator 1`] = `
<div
  class="t-urgent-callout border-8 rounded-bl-3xl py-10 pl-6 pr-8 sm:p-10 flex border-yellow-200 testClassName"
  data-testid="urgent-callout"
>
  <div
    class="flex gap-4 sm:gap-6"
  >
    <div
      class="text-pink-800"
    >
      <svgr-mock
        classname=""
      />
    </div>
    <div>
      children
    </div>
  </div>
</div>
`;

exports[`UrgentCallout component renders with a calculator and teal border 1`] = `
<div
  class="t-urgent-callout border-8 rounded-bl-3xl py-10 pl-6 pr-8 sm:p-10 flex border-teal-300 testClassName"
  data-testid="urgent-callout"
>
  <div
    class="flex gap-4 sm:gap-6"
  >
    <div
      class="text-pink-800"
    >
      <svgr-mock
        classname=""
      />
    </div>
    <div>
      children
    </div>
  </div>
</div>
`;

exports[`UrgentCallout component renders with a warning 1`] = `
<div
  class="t-urgent-callout border-8 rounded-bl-3xl py-10 pl-6 pr-8 sm:p-10 flex border-yellow-200 testClassName"
  data-testid="urgent-callout"
>
  <div
    class="flex gap-4 sm:gap-6"
  >
    <div
      class="text-pink-800"
    >
      <svgr-mock
        classname=""
      />
    </div>
    <div>
      children
    </div>
  </div>
</div>
`;

exports[`UrgentCallout component renders with a warning and teal border 1`] = `
<div
  class="t-urgent-callout border-8 rounded-bl-3xl py-10 pl-6 pr-8 sm:p-10 flex border-teal-300 testClassName"
  data-testid="urgent-callout"
>
  <div
    class="flex gap-4 sm:gap-6"
  >
    <div
      class="text-pink-800"
    >
      <svgr-mock
        classname=""
      />
    </div>
    <div>
      children
    </div>
  </div>
</div>
`;

exports[`UrgentCallout component renders with an arrow 1`] = `
<div
  class="t-urgent-callout border-8 rounded-bl-3xl py-10 pl-6 pr-8 sm:p-10 flex border-yellow-200 testClassName"
  data-testid="urgent-callout"
>
  <div
    class="flex gap-3 sm:gap-6"
  >
    <div
      class="text-pink-800 sm:hidden -ml-8"
    >
      <svgr-mock
        classname=""
      />
    </div>
    <div
      class="text-pink-800 hidden sm:block -mt-12 -ml-4"
    >
      <svgr-mock
        classname=""
      />
    </div>
    <div>
      children
    </div>
  </div>
</div>
`;

exports[`UrgentCallout component renders with an arrow and teal border 1`] = `
<div
  class="t-urgent-callout border-8 rounded-bl-3xl py-10 pl-6 pr-8 sm:p-10 flex border-teal-300 testClassName"
  data-testid="urgent-callout"
>
  <div
    class="flex gap-3 sm:gap-6"
  >
    <div
      class="text-pink-800 sm:hidden -ml-8"
    >
      <svgr-mock
        classname=""
      />
    </div>
    <div
      class="text-pink-800 hidden sm:block -mt-12 -ml-4"
    >
      <svgr-mock
        classname=""
      />
    </div>
    <div>
      children
    </div>
  </div>
</div>
`;
